File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.HashMap;
class Solution {
 public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        HashMap<Character,Integer> keys = new HashMap<>();
        for(int i = 0; i < keymap.length; i++){
            for(int j = 0; j < keymap[i].length(); j++){
                if(keys.containsKey(keymap[i].charAt(j))){
                    if(keys.get(keymap[i].charAt(j)) > j){
                        keys.put(keymap[i].charAt(j),j+1);
                    }
                }
                else{
                    keys.put(keymap[i].charAt(j),j+1);
                }
            }
        }
        for(int k = 0; k < targets.length; k++){
            for(int n = 0; n < targets[k].length(); n++){
                if(keys.containsKey(targets[k].charAt(n))){
                    int key = keys.get(targets[k].charAt(n));
                    answer[k] += key;
                }else{
                    answer[k] = -1;
                    break;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    hash_map = {}
    
    for key in keymap:
        for i in range(len(key)):
            if key[i] in hash_map:
                if hash_map[key[i]] > i + 1:
                    hash_map[key[i]] = i + 1
            else:
                hash_map[key[i]] = i + 1
    
    for target in targets:
        count = 0
        for j in range(len(target)):
            if target[j] in hash_map:
                count += hash_map[target[j]]
            else:
                count = -1
                break
                
        answer.append(count)
    
    return answer